board: Add power-on reset env variable

rajaganeshr opened the pull request:
Adds support to read syscontrol reset status register to check if BMC booting after AC power on. Exports the Boolean value in u-boot env var, so that BMC can read DIMM SPD.

chintingkuo committed:
arm: ast2700: Correct WDT reset information

Correct WDT reset check bit field.

Signed-off-by: Chin-Ting Kuo <chin-ting_kuo@aspeedtech.com>
Change-Id: I3c3c33cc2feb6579c4e1670100d2f59ff97696fc

mahkurap commented on board/aspeed/evb_ast2700/evb_ast2700.c:
```diff
@@ -233,6 +235,23 @@ void aspeed_errata_app_note_6_pcie_ep()
+
+       /* set reset reason env */
+       printf("Scratch register value: 0x%08x\n", por_rst);
+       if (por_rst & (u32)0x1)
```
you may want to define a macro for the scratch register bit mask and use it here.

rajaganeshr replied:
Done

mahkurap commented on arch/arm/mach-aspeed/ast2700/cpu-info.c:
```diff
@@ -47,9 +46,9 @@ void ast2700_print_soc_id(void)
 
 #define WDT_RST_BIT_MASK(s)	(GENMASK(3, 0) << (s))
-#define BIT_WDT_SOC(s)		(BIT(0) << (s))
+#define BIT_WDT_SOC(s)		(BIT(2) << (s))
```
how are these changes related to POR?

rajaganeshr replied:
Part of aspeed repo changes, cherrypicked origin commit in same pull request

mahkurap commented on arch/arm/mach-aspeed/ast2700/cpu-info.c:
```diff
@@ -83,10 +82,13 @@ void ast2700_print_wdtrst_info(void)
 void ast2700_print_sysrst_info(void)
 {
 	u32 sys_rst = readl(ASPEED_CPU_RESET_LOG1);
+	writel(sys_rst, ASPEED_SYS_SCRATCH_7FC);
```
why are we writing to 0x12C027FC?  You may want to add a sentence about this as a comment or in the commit log.

rajaganeshr replied:
Done

ojayanth commented on arch/arm/mach-aspeed/ast2700/cpu-info.c:
```diff
@@ -47,9 +46,9 @@ void ast2700_print_soc_id(void)
-#define BIT_WDT_ARM(s)		(BIT(2) << (s))
+#define BIT_WDT_ARM(s)		(BIT(0) << (s))
 #define BIT_WDT_SW(s)		(BIT(3) << (s))
 
```
Please make the above  two changes with cherry pick  https://github.com/AspeedTech-BMC/u-boot/commit/86500b3733c221e1287e86e80fb496704ba06873

rajaganeshr replied:
Done

rajaganeshr committed:
board: Add power-on reset env variable

Adds support to read syscontrol reset status register to check if BMC
booting after AC power on. Exports the boolean value in u-boot env var,
so that BMC can read DIMM SPD.

Signed-off-by: Rajaganesh Rathinasabapathi <Rajaganesh.Rathinasabapathi@amd.com>

ojayanth commented on board/aspeed/evb_ast2700/evb_ast2700.c:
```diff
@@ -233,6 +239,20 @@ void aspeed_errata_app_note_6_pcie_ep()
        printf("applying Aspeed app note 6 pcie ep ... Done\n");
 }
 
+void update_por_env(void)
```
OpenBMC design document https://github.com/openbmc/docs/blob/master/designs/bmc-reboot-cause-update.md provides the current way of BMC reset handling. In this document "Each BMC vendor needs to record the results of the retrieved reboot cause to the specified path." . and PSM consumes this . Can you look the driver based approch helps to meet AMD requirement?

rajaganeshr replied:
The proposal mentions 
> "Each BMC vendor must provide a driver to retrieve the BMC reboot cause and record the result at the specified location."

I guess we can merge u-boot approach for now and later migrate to when Aspeed provides a driver.

ojayanth replied:
As discussed , will use this approch for now , till the aspeed kernel support is available.
